src/game.py

```diff
@@ -0,0 +1,284 @@
+        """
+        spots_empty = 1
+        for number in numbers:
+            bit = 1 << number
+            spots = []
+            spots = self.move_for_y_position(allowed, x, axis, bit, spots)
+            if len(spots) == 0:
+                spots_empty = 0
+            elif len(spots) == 1:
+                board[spots[0]] = number
+                stuck = False
+            elif stuck:
+                guess, count = self.pick_better(guess, count, [(position, number) for position in spots])
+        return spots_empty, board, stuck, guess, count
+
+    def move_for_y_position(self, allowed, x, axis, bit, spots):
+        """Move on axis position for each axis y  and guess allowed values
+
+        Keyword arguments:
+        allowed -- Array of numbers allowed to resolve puzzle
+        x  -- Number from 1 to 9
+        axis -- Number for each iteration of move_on_board_for_axis function
+        bit -- get a 0 or 1 for flag deduction
+        spots -- It has a candidate numbers to be chosen
+        return  -- a list os spots for allowed values
+        """
+        for y in xrange(9):
+            position = self.set_position_for(x, y, axis)
+            if allowed[position] & bit:
+                spots.append(position)
+        return spots
+
+    def move_on_board(self, board, allowed, stuck, guess, count):
+        """Move on board for each position from 1 to 81
+
+        Keyword arguments:
+        allowed -- Array of numbers allowed to resolve puzzle
+        board -- Array e.g.: [[None, 1,..., 8, 6]]
+        guess -- Array of random numbers from 1 to 9
+        count -- Count how many times pick the best option of guesses
+        stuck -- Verify if game stuck returning true or false
+        return  -- Tuple of (board, allowed, stuck, guess, count)
+        """
+        for position in xrange(81):
+            result = self.verify_board(position, board, allowed, stuck, guess, count)
+            if result[0] == 0:
+                return []
+            if result[0] == 1:
+                board = result[1]
+                stuck = result[2]
+            if result[0] == 2:
+                guess = result[1]
+                count = result[2]
+        return board, allowed, stuck, guess, count
+
+    def verify_board(self, position, board, allowed, stuck, guess, count):
+        """Move on board for each position from 1 to 81
+
+        Keyword arguments:
+        position -- Number of board position
+        allowed -- Array of numbers allowed to resolve puzzle
+        board -- Array e.g.: [[None, 1,..., 8, 6]]
+        guess -- Array of random numbers from 1 to 9
+        count -- Count how many times pick the best option of guesses
+        stuck -- Verify if game stuck returning true or false
+        axis -- Number for each iteration of move_on_board_for_axis function
+        return  -- Tuple of (zero, board, stuck, guess, count) where zero is a flag
```

Comment:
again these flags need to be well documented specially when they are not boolean values